src/ethereum/base_types.py

```diff
@@ -636,4 +637,69 @@ def to_signed(self) -> int:
 Bytes20 = Bytes
 Bytes32 = Bytes
 Bytes64 = Bytes
-BytesMutable256 = bytearray
+Bytes256 = Bytes
+
+
+def _setattr_function(self: Any, attr: str, value: Any) -> None:
+    if self._frozen:
+        raise Exception("Mutating frozen dataclasses is not allowed.")
+    else:
+        object.__setattr__(self, attr, value)
+
+
+def _delattr_function(self: Any, attr: str) -> None:
+    if self._frozen:
+        raise Exception("Mutating frozen dataclasses is not allowed.")
+    else:
+        object.__delattr__(self, attr)
+
+
+def _make_init_function(f: Callable) -> Callable:
+    def init_function(self: Any, *args: Any, **kwargs: Any) -> None:
+        will_be_frozen = kwargs.pop("_frozen", True)
+        object.__setattr__(self, "_frozen", False)
+        f(self, *args, **kwargs)
+        self._frozen = will_be_frozen
+
+    return init_function
+
+
+def slotted_freezable(cls: Any) -> Any:
```

Comment:
I assume the mypy type checks still work with decorated classes?

Reply:
mypy doesn't understand `@slotted_freezable` and treats it as a no-op. The only problem with this is mypy will incorrectly complain if you try to access the `_frozen` element.

Otherwise everything works fine.